src/main/java/com/gamebros/purepazaak/Match.java

```diff
@@ -0,0 +1,81 @@
+package com.gamebros.purepazaak;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.gamebros.purepazaak.entity.Card;
+import com.gamebros.purepazaak.entity.Player;
+import com.gamebros.purepazaak.enumerable.PlayerEnum;
+import com.gamebros.purepazaak.enumerable.WinnerEnum;
+
+public class Match {
+  protected Player playerOne;
+
+  protected Player playerTwo;
+
+  protected ArrayList<Card> playerOneDeck;
+
+  protected ArrayList<Card> playerTwoDeck;
+
+  protected ArrayList<MatchSet> sets = new ArrayList<MatchSet>();
+
+  protected MatchSet currentSet;
+
+  public Match(Player playerOne, Player playerTwo) {
+    this.playerOne = playerOne;
+    this.playerTwo = playerTwo;
+
+    this.playerOneDeck = playerOne.getSideDeck().drawRandomCards();
+    this.playerTwoDeck = playerTwo.getSideDeck().drawRandomCards();
+  }
+
+  public MatchSet newSet() {
+    MatchSet set = new MatchSet(new MainDeck());
+
+    this.sets.add(set);
+    this.currentSet = set;
+
+    return set;
+  }
+
+  public ArrayList<MatchSet> getMatchSets() {
+    return this.sets;
+  }
+
+  public MatchSet getCurrentSet() {
+    return this.currentSet;
+  }
+
+  public Player getWinner() {
+    if (this.sets.size() < 3) {
+      return null;
+    }
+
+    ArrayList<MatchSet> playerOne = this.sets
```

Comment:
I don't think it benefits the domain model to move this logic to a separate class.  This class is the only class concerned with the details of deciding the winner, therefore it will never need to change.